arbor/include/arbor/morph/mprovider.hpp

```diff
@@ -20,6 +22,7 @@ struct ARB_ARBOR_API mprovider {
     // Throw exception on missing or recursive definition.
     const mextent& region(const std::string& name) const;
     const mlocation_list& locset(const std::string& name) const;
+    const std::shared_ptr<iexpr_interface>& iexpr(const std::string& name) const;
```

Comment:
Maybe at this point add `iexpr_ptr = std::shared_ptr<iexpr_interface>`
(other places in arbor do it this way)